### PR TITLE
Bug solved

### LINKED ISSUE #613: pc.html does not pay attention to tags
**Describe the bug**
header tags like `<h1></h1>` have no effect on the text between the tabs

**To Reproduce**
you can just look at your own docs on website
 - Code/Link to Repo:](https://pynecone.io/docs/library/other/html)

**Expected behavior**
header tags should produce bolded and englarged text

**Screenshots**

** Specifics (please complete the following information):**
 - Python Version: 3.9
 - Pynecone Version: latest
 - OS:  mac
 - Browser (Optional): safari, chrome
 
**Additional context**

### PR DESCRIPTION
**Pull request by danik292**
Reviving a PR from issue #613.
Closes #613

### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### **After** these steps, you're ready to open a pull request.

    a. Give a descriptive title to your PR.

    b. Describe your changes.

    c. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Comment by danik292** [comment]
@Alek99 hello please check this PR

File: reflex/components/layout/html.py [hunk 1]
```diff
@@ -31,9 +31,35 @@ def create(cls, *children, **props):
...
+            raise ValueError("Must provide a single child to the html component.")
         else:
-            props["dangerouslySetInnerHTML"] = {"__html": children[0]}
+            child = children[0]
+            wrapped_child = child # Initialize with the child as is
+
+            # Check for header tags and wrap them accordingly
+            if child.startswith("<h1>") and child.endswith("</h1>"):
+                wrapped_child = f"<h1>{child[4:-5]}</h1>"
+                elif child.startswith("<h2>") and child.endswith("</h2>"):
```
Comment (masenf) [inline reflex/components/layout/html.py:42]: indentation is broken here

```console
  File "/Users/masen/code/reflex-dev/reflex/reflex/components/layout/html.py", line 42
    elif child.startswith("<h2>") and child.endswith("</h2>"):
    ^^^^
SyntaxError: invalid syntax
```

**Review by masenf — CHANGES_REQUESTED** [review]
This doesn't seem to work for me

`rx.html("<h1>foo</h1><h2>bar</h2>")` is formatted like 
image

I think the problem here isn't about wrapping the tag, but more about how the styles are applied.

If i use that same snippet against reflex 0.2.3, but uncheck the CSS properties, it renders as you'd expect

image
image

For the framework, we would need a more general solution that works for arbitrary HTML instead of special casing where the child string starts and ends with the same tag.

**Comment by masenf** [comment]
In tailwind and chakra the unstyled headings are by design: https://tailwindcss.com/docs/preflight#headings-are-unstyled

But for uncontrolled html, you might try the tailwind typography plugin: https://tailwindcss.com/docs/typography-plugin

in `rxconfig.py`:

```python
import reflex as rx

config = rx.Config(
    app_name="repro_event_yield_return",
    tailwind={
        "plugins": ["@tailwindcss/typography"],
    },
    frontend_packages=["@tailwindcss/typography"],
)
```

then in my frontend I can apply the `class_name="prose"` to the `rx.html` component and get "normal" rendering

```python
def index():
    return rx.html("<h1>Foo</h1><h2>Bar</h2><p>This text is <em>italic</em> and this is <strong>bold</strong>", class_name="prose")
```

image


See https://reflex.dev/docs/styling/overview/#tailwind

**Comment by masenf** [comment]
Filed #1529 to track a more general fix

**Comment by danik292** [comment]
So I helped us?

**Pull request closed without merging**